unitTests/rest/BackOffRequestTest.ts

```diff
@@ -0,0 +1,65 @@
+import { BackOffRequest, setBackOffModule } from '../../src/rest/BackOffRequest';
+import { ISuccessResponse } from '../../src/rest/EndpointCaller';
+
+export function BackOffRequestTest() {
+  describe('BackOffRequest', () => {
+    const oneTimeUnit = 100;
+    const mockSuccessResponse: ISuccessResponse<{}> = {
+      duration: 1,
+      data: {}
+    };
+
+    beforeAll(() => stubBackOffDependency());
+
+    function stubBackOffDependency() {
+      setBackOffModule(fn => fn());
+    }
+
+    function resolveInOneTimeUnit(): Promise<ISuccessResponse<{}>> {
+      return new Promise(resolve => setTimeout(() => resolve(mockSuccessResponse), oneTimeUnit));
+    }
+
+    function roundToNearestTimeUnit(duration: number) {
+      const numOfUnits = Math.round(duration / oneTimeUnit);
+      return numOfUnits * oneTimeUnit;
+    }
+
+    it(`when calling #enqueue with a request that resolves successfully,
+    it returns the successful response`, done => {
+      const request = BackOffRequest.enqueue({ fn: () => Promise.resolve(mockSuccessResponse) });
+
+      request.then(response => {
+        expect(response).toBe(mockSuccessResponse);
+        done();
+      });
+    });
+
+    it(`when calling #enqueue with two requests that resolve in one time unit each,
+    it processes the requests in series`, done => {
+      const startTime = Date.now();
+
+      BackOffRequest.enqueue({ fn: () => resolveInOneTimeUnit() });
+      const secondRequest = BackOffRequest.enqueue({ fn: () => resolveInOneTimeUnit() });
+
+      secondRequest.then(response => {
```

Comment:
Could we have a test that does something like this ?

```
const firstRequest = ()=> resolveIn5TimeUnits();
const secondRequest = ()=> resolveInOneTimeUnits();

firstRequest.then(()=> {
expect second request to not be resolved
})
```

And then another one where you add a `retryCondition` on the first and second request, and you check that the retry condition on the second request is not called before the first request is finished ?